app/controllers/admin/api/cms/templates_controller.rb

```diff
@@ -170,12 +168,9 @@ def find_section
     scope.find_by_id(params[:section_id]) || scope.find_by_system_name(params[:section_name]) || scope.root
   end
 
-  def set_layout_by(attr_name, finder, attrs)
-    if attrs.key?(attr_name)
-      attrs[:layout] = if name = attrs[attr_name].presence
-        current_account.layouts.send(finder,name)
-                       end
-    end
+  def find_layout
+    scope = current_account.layouts
+    scope.find_by_id(params[:layout_id]) || scope.find_by_system_name(params[:layout_name])
```

Comment:
We're querying the DB even when `layout_id` and `layout_name` are nil (in fact, if none of them is provided, we're still adding 2 extra queries), which is not nice, I think.

Reply:
But the original implementation is indeed confusing, so I am up for refactoring.

Reply:
I now also noticed the same logic already existed for sections (in `find_section`), I think we can change it also.

Reply:
Done!